docs/api.rst

```diff
@@ -80,26 +79,28 @@ Becomes using HPy API:
 Calling any HPy function on a closed handle is an error. Calling
 ``HPy_Close()`` on the same handle twice is an error. Forgetting to call
 ``HPy_Close()`` on a handle results in a memory leak. When running in
-:ref:`debug-mode:debug mode`, HPy actively checks that you that you don't close a handle
-twice and that you don't forget to close any.
+:ref:`debug-mode:debug mode`, HPy actively checks that you that you don't
+close a handle twice and that you don't forget to close any. This is
+possible because handles are unique - the handle used to reference an
+object is valid only in the context of the operation that produced it.
```

Comment:
I don't know if `in the context of the operation that produced it` isn't simplifying it bit too much or if it cannot be bit misleading?